libs/portable/generate.py

```
@@ -15,15 +15,27 @@
 # output: {path: (compressed_data, file_md5)}
 
 
-def generate_md5_table(folder: str, level) -> dict:
+def normalize(path: str) -> str:
+    path = path.replace('\\', '/')
+    while path.startswith('./'):
+        path = path[2:]
+    return path.lower()
+
+
+def generate_md5_table(folder: str, level, exclude: str = None) -> dict:
```

Ruff (0.16.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)